src/ast/pass_manager.cpp

```diff
@@ -1,47 +1,75 @@
 #include "ast/pass_manager.h"
-
-#include <ostream>
-
-#include "ast/passes/printer.h"
-#include "bpftrace.h"
+#include "ast/context.h"
+#include "log.h"
+#include "util/result.h"
 
 namespace bpftrace::ast {
 
-namespace {
-void print(PassContext &ctx, const std::string &name, std::ostream &out)
+std::atomic<int> PassContext::next_type_id_;
```

Comment:
JC, do we have multiple threads accessing this or is it just for future proofing?

Reply:
Static initializers have historically been fraught. I know that you have thread-safe guarantees for the *same* static initializer (multiple threads hitting the same static initializer, magically one goes and others block). You *may* have hard guarantees around concurrency for *different* static initializers, but I just prefer to be defensive for this case.